scripts/meterian-docker

```diff
@@ -21,60 +21,46 @@ else
-    NODE_LOCAL_DIR=$(npm root -g)
-    NODE_DOCKER_DIR=/usr/lib/node_modules
-    
-    # python specific mappings
-    LOCALPRG_LOCAL_DIR=~/.local
-    LOCALPRG_DOCKER_DIR=/home/meterian/.local
+    CONTAINER_HOME_DIR=/home/meterian
+    docker_run_command="docker run -it --rm --volume $WORKDIR:/workspace "
 
+    declare -A VOLUME_BINDS
+    VOLUME_BINDS=(
+        ["$HOME/.cache"]=$CONTAINER_HOME_DIR/.cache
+        ["$HOME/.local"]=$CONTAINER_HOME_DIR/.local
+    # maven specific mappings
+        ["$HOME/.m2"]=$CONTAINER_HOME_DIR/.m2
+    # dotnet specific mappings                            
+        ["$HOME/.dotnet"]=$CONTAINER_HOME_DIR/.dotnet
+        ["$HOME/.nuget"]=$CONTAINER_HOME_DIR/.nuget
+        # ruby specific mappings
+        [$(gem environment gemdir)/cache]=/var/lib/gems/2.3.0/cache
+        # node specific mappings
+        [$(npm root -g)]=/usr/lib/node_modules
     # gradle specific mappings
-    GRADLE_LOCAL_DIR=~/.gradle
-    GRADLE_DOCKER_DIR=/home/meterian/.gradle
-
+        ["$HOME/.gradle"]=$CONTAINER_HOME_DIR/.gradle
     # sbt specific mappings
-    SBT_GLOBAL_SETTINGS_LOCAL_DIR=~/.sbt
-    SBT_GLOBAL_SETTINGS_DOCKER_DIR=/home/meterian/.sbt
-    SBT_LOCAL_DIR=~/.ivy2
-    SBT_DOCKER_DIR=/home/meterian/.ivy2
-    
-    # executing docker now
-    docker run -it --rm \
-              --workdir ${WORKDIR} \
-              --volume ${WORKDIR}:${WORKDIR} \
-              --volume ${MAVEN_LOCAL_DIR}:${MAVEN_DOCKER_DIR} \
-              --mount type=bind,source=${DOTNET_LOCAL_DIR},target=${DOTNET_DOCKER_DIR} \
-              --mount type=bind,source=${CACHES_LOCAL_DIR},target=${CACHES_DOCKER_DIR} \
-              --mount type=bind,source=${RUBY_LOCAL_DIR},target=${RUBY_DOCKER_DIR} \
-              --mount type=bind,source=${NODE_LOCAL_DIR},target=${NODE_DOCKER_DIR} \
-              --mount type=bind,source=${LOCALPRG_LOCAL_DIR},target=${LOCALPRG_DOCKER_DIR} \
-              --mount type=bind,source=${GRADLE_LOCAL_DIR},target=${GRADLE_DOCKER_DIR} \
-              --mount type=bind,source=${SBT_GLOBAL_SETTINGS_LOCAL_DIR},target=${SBT_GLOBAL_SETTINGS_DOCKER_DIR} \
-              --mount type=bind,source=${SBT_LOCAL_DIR},target=${SBT_DOCKER_DIR} \
-              --env HOST_UID=${HOST_UID} \
-              --env METERIAN_API_TOKEN=${METERIAN_API_TOKEN} \
-              ${DOCKER_FULL_IMAGE_NAME} $*
+        ["$HOME/.sbt"]=$CONTAINER_HOME_DIR/.sbt
+        ["$HOME/.ivy2"]=$CONTAINER_HOME_DIR/.ivy2
+    )
+
+    for local_addr in "${!VOLUME_BINDS[@]}"; do 
+        if [ -d $local_addr ];
+        then
+            docker_addr=${VOLUME_BINDS[$local_addr]}
+            docker_run_command="$docker_run_command --mount type=bind,source=$local_addr,target=$docker_addr"
+        fi
+    done
+
+    docker_run_command="$docker_run_command --env HOST_UID=$HOST_UID --env HOST_GID=$HOST_GID"
+    docker_run_command="$docker_run_command --env METERIAN_API_TOKEN=$METERIAN_API_TOKEN $DOCKER_FULL_IMAGE_NAME \$*"
+    eval $docker_run_command
```

Comment:
I would make sure it's impossible to tamper with the input variables, as the "eval" will evaluate anything in the middle. Is there a way to make sure any possibly harmful content present in the variable is escaped?
See here for a possible strategy:
https://unix.stackexchange.com/questions/379181/escape-a-variable-for-use-as-content-of-another-script

Reply:
Resolved with this [commit](https://github.com/MeterianHQ/meterian-scanner-docker/pull/10/commits/3f3d6165c4655107f2884fae9b9a760fa3f2a7b2).